In [19]:
import numpy as np
import pandas as pd

1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):  
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],  
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].  
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны.

In [38]:
def cov_manual(arr1, arr2):
    if len(arr1) != len(arr2):
        return None
    else:
        return sum(arr1 * arr2) / len(arr1) - sum(arr1)/len(arr1) * sum(arr2) / len(arr2)
    
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

cov_manual(zp, ks), np.cov(zp, ks, bias=True)[0,1]

(9157.839999999982, 9157.84)

___

2. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [45]:
zp_ser = pd.Series(zp)
ks_ser = pd.Series(ks)

cov_manual(zp, ks) / (zp.std() * ks.std()), np.corrcoef(zp, ks)[0,1], zp_ser.corr(ks_ser)

(0.8874900920739144, 0.8874900920739162, 0.8874900920739162)

___

3. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:  
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.  
Известно, что в генеральной совокупности IQ распределен нормально.  
Найдите доверительный интервал для математического ожидания с надежностью 0.95.  

In [2]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
iq.mean(), len(iq), iq.std()

(118.1, 10, 10.004498987955369)

$ \bar{X}-t_{\alpha=0.95, n=9}\cdot\frac{\sigma_x}{\sqrt{n}}<\mu<\bar{X}+t_{\alpha=0.95, n=9}\cdot\frac{\sigma_x}{\sqrt{n}} $  

$ \bar{X} = 118.1 $  
$ \sigma_x = 10.005 $  
$ n = 10 $  
$ t_{\alpha=0.95, n=9}=2.262 $  
$ 118.1 - 2.262 \cdot \frac{10.005}{\sqrt{10}}<\mu<118.1 + 2.262 \cdot \frac{10.005}{\sqrt{10}} \Rightarrow $  

$$ 110.94<\mu<125.26 $$



___

4. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

$ \sigma^2 = 25 $  
$ n = 27 $  
$ \bar{X} = 174.2 $  
$ \alpha = 0.05 $  

$ \sigma = 5 $  
$ t_{\alpha=0.95, n=26}=2.056 $  

$ 174.2 - 2.056 \cdot \frac{5}{\sqrt{27}}<\mu<174.2 + 2.056 \cdot \frac{5}{\sqrt{27}} \Rightarrow $  

$$ 172.22<\mu<176.18 $$